src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -476,6 +476,11 @@ public DType GetType(TexlNode node)
             Contracts.AssertIndex(node.Id, _typeMap.Length);
             Contracts.Assert(_typeMap[node.Id].IsValid);
 
+            return GetTypeAllowInvalid(node);
+        }
+
+        public DType GetTypeAllowInvalid(TexlNode node)
```

Comment:
Fir for #2414 - code that can handle invalid types can call opt into calling this method.